pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -2103,8 +2103,10 @@ public void redeliverUnacknowledgedMessages() {
                 incomingQueueLock.unlock();
             }
 
-            // is channel is connected, we should send redeliver command to broker
-            if (cnx != null && isConnected(cnx)) {
+            // If a subscription command has been sent to the broker, it is necessary to allow the redelivery
+            // request to be sent to the broker without checking the connection state, as failing to do so would
+            // result in the client consumer epoch being bigger than the broker consumer epoch.
```

Comment:
Why does the connection state check need to be skipped? It's mentioned in the comment, but I don't understand the behavior.

Reply:
- client consumer send reconnect command to broker  epoch = 1
- client consumer invoke redeliverUnacknowledgedMessages epoch = 2
- if check connect, client consumer will not send redeliverUnacknowledgedMessages command with epoch = 2 to broker
- client consumer epoch = 2 will bigger than broker consumer epoch = 2, client will filter all messages with epoch = 1